### PR TITLE
Added pre-validation check playbook

### PR DESCRIPTION
**Pull request by mwitzenm**
#### What is this PR About?
Added a pre-validation delivery playbook

#### How should we test or review this PR?
Please review and suggest changes

#### Is there a relevant Trello card or Github issue open for this?
https://trello.com/c/yTqV09nu/23-create-a-pre-validation-delivery-runbook

#### Who would you like to review this?
cc: @redhat-cop/cant-contain-this

### CODE REVIEW

File: playbooks/installation/pre-validation.adoc [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+v1.0, 2017-03-29
+:toc: macro
+:toc-title:
+
+toc::[]
+
+
+== Ensure ssh keys are propagated for ansible installer
+
+* The Ansible installer requires either:
```
Comment (etsauer) [inline playbooks/installation/pre-validation.adoc]: Should not be a list item

File: playbooks/installation/pre-validation.adoc [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+* root password-less ssh access using ssh keys 
+* A non-root user with password-less ssh acess and full password-less sudo access 
+
+----
+How to Propagate your key example:
```
Comment (etsauer) [inline playbooks/installation/pre-validation.adoc]: This isn't code. I would move this outside of the code block.

File: playbooks/installation/pre-validation.adoc [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+ssh-copy-id -i ~/.ssh/Myidrsa.pub remote.server.com
+----
+
+== DNS lookup on each node in your cluster
+
+* This is to ensure dns is properly set
+
+----
+example:
```
Comment (etsauer) [inline playbooks/installation/pre-validation.adoc]: move out of code block

File: playbooks/installation/pre-validation.adoc [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+* A non-root user with password-less ssh acess and full password-less sudo access 
+
+----
+How to Propagate your key example:
+ssh-copy-id -i ~/.ssh/Myidrsa.pub remote.server.com
+----
+
+== DNS lookup on each node in your cluster
+
+* This is to ensure dns is properly set
```
Comment (etsauer) [inline playbooks/installation/pre-validation.adoc]: single statement doesn't need a bullet

File: playbooks/installation/pre-validation.adoc [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+----
+example:
+dig www.google.com +short
+172.217.7.132
+----
+
+== DNS reverse lookup on each node 
+
+----
+example:
```
Comment (etsauer) [inline playbooks/installation/pre-validation.adoc]: move out of code block

File: playbooks/installation/pre-validation.adoc [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+dig -x `dig www.google.com +short` +short
+iad23s59-in-f4.1e100.net.
+----
+
+== OCP 3.3 install repos
+
+* If Installing OpenShift 3.3 please use these repos
+
+----
+subscription-manager repos \
```
Comment (etsauer) [inline playbooks/installation/pre-validation.adoc]: if this is a pre-validation test, I wouldn't include commands that can change the state of things. Can you change this to do a yum repolist and validate the right repos show up?

File: playbooks/installation/pre-validation.adoc [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+    --enable="rhel-7-server-optional-rpms"\
+    --enable="rhel-7-server-ose-3.3-rpms"
+----
+
+== OCP 3.4 install repos
+
+* If Installing OpenShift 3.4 please use these repos
+
+----
+subscription-manager repos \
```
Comment (etsauer) [inline playbooks/installation/pre-validation.adoc]: same here. don't want to change state in a pre-flight script

File: playbooks/installation/pre-validation.adoc [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+subscription-manager status
+----
+
+== To list all repos recognized by your rhel 7 nodes
+
+----
+subscription-manager repos
+----
+
+== Install some extras required for OpenShift
```
Comment (etsauer) [inline playbooks/installation/pre-validation.adoc]: This isn't necessary for a pre-install validation, as the installer will install these. I don't want to blur the lines between validation, and prepping hosts.

File: playbooks/installation/pre-validation.adoc [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+
+----
+yum install -y wget git net-tools bind-utils iptables-services bridge-utils bash-completion atomic-openshift-utils docker-selinux docker
+----
+
+== Ensure docker is enabled and started
```
Comment (etsauer) [inline playbooks/installation/pre-validation.adoc]: same idea here. we should check whether docker has been enabled, but shouldn't change the state.

File: playbooks/installation/pre-validation.adoc [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+
+----
+systemctl enable docker
+systemctl start docker
+----
+
+== Ensure NetworkManager is enabled and started
```
Comment (etsauer) [inline playbooks/installation/pre-validation.adoc]: don't change state.

File: playbooks/installation/pre-validation.adoc [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+Mode from config file:          enforcing
+Policy MLS status:              enabled
+Policy deny_unknown status:     allowed
+Max kernel policy version:      28
+
+grep ^SELINUX= /etc/sysconfig/selinux  
+SELINUX=enforcing
+----
+
+== Set SELinux nfs write booleans on all nodes in the cluster
```
Comment (etsauer) [inline playbooks/installation/pre-validation.adoc]: This is system prep, not validation.

File: playbooks/installation/pre-validation.adoc [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+---
+= OpenShift Pre-Install Validation Checklist
+Matt Witzenman <mwitzenm@redhat.com>
+v1.0, 2017-03-29
+:toc: macro
+:toc-title:
+
+toc::[]
+
+
```
Comment (etsauer) [inline playbooks/installation/pre-validation.adoc]: Can we give an overview of what this document is? and set some context a bit?

File: playbooks/installation/pre-validation.adoc (hunk 8 above)
Comment (mwitzenm) [inline playbooks/installation/pre-validation.adoc]: I am confused ...I did not realize the installer actually installs everything it needs?

**Comment by JaredBurck** [comment]
reviewed...LGTM!

**Pull request merged**